stream.go

```diff
@@ -217,21 +217,28 @@ func (s *StreamService) Dial(ctx context.Context, params *DialStreamParams) (*Co
 
 	go func() {
 		defer conn.Close()
+		var prevErr error = nil
 		for {
 			var msg StreamMessage
 			if err := conn.conn.ReadJSON(&msg); err != nil {
 				if e, ok := err.(*websocket.CloseError); ok {
 					if isRevError, revError := IsRevError(e.Code); isRevError {
 						conn.err <- revError
-						return
+					} else {
+						conn.err <- e
 					}
+					// if we recieve any CloseError the connection is closed and needs to be reestablished before reading can continue
+					return
 				}
 
-				if websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway, websocket.CloseAbnormalClosure) {
-					// perhaps an error should be sent on Err here too
+				// ReadJson either returns a json decode error of will return the error again and again
+				// eventually leading to a panic. if we get the same error repeatedly report and finish.
+				if err == prevErr {
 					conn.err <- err
 					return
 				}
+
+				// silently drop read error.
 				continue
```

Comment:
Currently errors such as a json decode error or similar are not reported back to the calling application, Ideally they should be communicated so any recurring issues can be addressed.